rust/tw_bitcoin/src/modules/signer.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{Error, Result};
+use bitcoin::key::{TapTweak, TweakedKeyPair};
+use bitcoin::sighash::{EcdsaSighashType, TapSighashType};
+use secp256k1::{KeyPair, Message, Secp256k1};
```

Comment:
I think it's worth to replace `secp256k1` with `k256` that we use in `tw_key_pair`. But after the security audit

Reply:
Ideally yes, but Taproot introduces some complexity here, specifically: https://docs.rs/bitcoin/latest/bitcoin/key/trait.TapTweak.html#tymethod.tap_tweak. I can look into it and determine whether this is hard to do ourselves with `k256`.